### ISSUE TITLE
compilation error with cygwin 2.738

### ISSUE THREAD

**Issue by ghost** [SFML/SFML#140]
I was able to compile a Makefile using the cmake provided by cygwin; however, it appears that I'm getting an error message from make stating that "This operating system is not supported by SFML library."  It appears that cygwin is not being mapped as a supportable platform.

**Comment by kaeptmblaubaer1000** [comment]
SFML tries to use _beginthreadex and _endthreadex but I have set __linux__.
Cygwin doesn`t provides _beginthreadex and _endthreadex.
SFML must use w32api or pthread for that.

**Comment by MarioLiebisch** [comment]
Could you elaborate on how you did that? SFML won't magically think you're on Linux just because you define `__linux`. Can you provide step by step instructions?

**Comment by kaeptmblaubaer1000** [comment]
1. Open include/SFML/Config.hpp
2. Insert `#define __linux__` at the top
3. Run make

**Comment by MarioLiebisch** [comment]
That won't work –  you can't just define that and expect it to work.

Are these the only two functions that are missing/failing?

**Comment by MarioLiebisch** [comment]
Okay, had a look, and with some minor modifications I got the SFML stuff to compile (mostly), but then I've got tons of problems within Cygwin's headers itself (related to pthreads etc. despite me not using pthreads), so… without wanting to sound negative, this looks like too much trouble just to support some weird edge case most probably won't ever need.

If you want to use the GNU compiler and stuff on Windows, use MinGW. Just install it to "C:\usr" and most paths will play out almost like you're used to on Linux systems.

If anyone feels investigating this further, feel free to do so, but I think there are significantly more important issues right now. :)
